extensions/terminal-suggest/src/terminalSuggestMain.ts

```diff
@@ -106,26 +107,17 @@ export async function activate(context: vscode.ExtensionContext) {
 				return undefined;
 			}
 
-			const uniqueResults = new Map<string, vscode.TerminalCompletionItem>();
-			for (const item of items) {
-				if (!uniqueResults.has(item.label)) {
-					uniqueResults.set(item.label, item);
-				}
-			}
-			const resultItems = uniqueResults.size ? Array.from(uniqueResults.values()) : undefined;
-
-			// If no completions are found, the prefix is a path, and neither files nor folders
+			// If the command line is empty, no completions are found, or the completion found is '.', the prefix is a path, and neither files nor folders
 			// are going to be requested (for a specific spec's argument), show file/folder completions
-			const shouldShowResourceCompletions = !resultItems?.length && prefix.match(/^[./\\ ]/) && !filesRequested && !foldersRequested;
+			const shouldShowResourceCompletions = (terminalContext.commandLine.trim().length === 0 || !items?.length || items.length === 1 && items[0].label === '.') && (!filesRequested && !foldersRequested);
```

Comment:
nit: you could move the comments closer to the conditions for harder to parse code like this:

```
const shouldShowResourceCompletions = (
  // Command line is empty
  terminalContext.commandLine.trim().length === 0 ||
  // ...
  !items?.length ||
  ...
);
```